akka-stream/src/main/scala/akka/stream/impl/fusing/ActorGraphInterpreter.scala

```diff
@@ -812,23 +816,31 @@ import akka.util.OptionVal
     }
   }
 
-  override def receive: Receive = {
-    case b: BoundaryEvent =>
-      currentLimit = eventLimit
-      processEvent(b)
-      if (shortCircuitBuffer != null) shortCircuitBatch()
-
-    case Resume =>
-      currentLimit = eventLimit
-      if (shortCircuitBuffer != null) shortCircuitBatch()
-
-    case Snapshot =>
-      sender() ! StreamSnapshotImpl(
-        self.path,
-        activeInterpreters.map(shell => shell.toSnapshot.asInstanceOf[RunningInterpreter]).toSeq,
-        newShells.map(shell => shell.toSnapshot.asInstanceOf[UninitializedInterpreter]))
+  override protected[akka] def aroundReceive(receive: Receive, msg: Any): Unit = {
+    // as we know we never become in stream actors, we can avoid some overhead with the partial function/behavior
+    // stack of classic actors entirely we also know that the receive is total, so we can avoid the orElse part as well.
+    msg match {
+      case b: BoundaryEvent =>
+        currentLimit = eventLimit
+        processEvent(b)
+        if (shortCircuitBuffer != null) shortCircuitBatch()
+
+      case Resume =>
+        currentLimit = eventLimit
+        if (shortCircuitBuffer != null) shortCircuitBatch()
+
+      case Snapshot =>
+        sender() ! StreamSnapshotImpl(
+          self.path,
+          activeInterpreters.map(shell => shell.toSnapshot.asInstanceOf[RunningInterpreter]).toSeq,
+          newShells.map(shell => shell.toSnapshot.asInstanceOf[UninitializedInterpreter]))
+
+      case other => unhandled(other)
+    }
   }
 
+  override def receive: Receive = ActorGraphInterpreter.DummyReceive
```

Comment:
Is there any reason not inline the value definition?

Reply:
Actually only called once, during actor startup, so not important to optimize